### PR TITLE
fix(macro): Add keystore extension to vtk objects

### PR DESCRIPTION
**Pull request by floryst**

### CODE REVIEW

**Comment by floryst** [comment]
The keystore is basically an ES6 Map. (ES5? I forget) Do we want to just forward all Map methods?

File: Sources/macro.js [hunk 1]
```diff
@@ -1200,6 +1200,37 @@ export function proxyPropertyState(
...
+//
+// Generated API
+//  setKey(key, value) : undefined
+//  getKey(key) : mixed
+//  getAllKeys() : [mixed]
+//  deleteKey(key) : Boolean
+// ----------------------------------------------------------------------------
+
+export function keystore(publicAPI, model, initialKeystore = {}) {
+  const store = new Map();
```
Comment (jourdain) [inline Sources/macro.js]: Do we really need a Map? Would it be good enough to use `{}` instead?

Do we want it stored on the `model`?

**Comment by jourdain** [comment]
Then we should add a call to it inside the macro.proxy

File: Sources/macro.js (hunk 1 above)
Comment (floryst) [reply Sources/macro.js]: I tend to use Map for key/value stores. Plus, if someone really wants to use an object as a key, they can with Map. :P Is there a reason to use `{}` instead?

Yes, storing on the model would be nice. I'll do that.

Comment (jourdain) [reply Sources/macro.js]: I understand, but then you break the current pattern that enable serialization on any vtkObject:

```
const str = obj.getState();
const newObj = vtk(str);
```

Also you would have to do something like

```
if (!model.keystore) {
   model.keystore = {};
}
```

Comment (floryst) [reply Sources/macro.js]: Oh, didn't know about object serialization. In that case yes, I'll switch to using an object.

**Pull request merged**